braindecode/datasets/base.py

```diff
@@ -0,0 +1,62 @@
+"""
+Dataset classes.
+"""
+
+# Authors: Hubert Banville <hubert.jbanville@gmail.com>
+#          Lukas Gemein <l.gemein@gmail.com>
+#          Simon Brandt <simonbrandt@protonmail.com>
+#          David Sabbagh <dav.sabbagh@gmail.com>
+#
+# License: BSD (3-clause)
+
+from torch.utils.data import Dataset
+
+
+class BaseDataset(Dataset):
+    """
+    A base dataset.
+
+    Parameters
+    ----------
+    raw: mne.Raw
+    info: pandas.DataFrame
+        holds additional information about the raw
+    """
+    def __init__(self, raw, info, target=None):
+        self.raw = raw
+        self.info = info
+        if target is not None:
+            assert target in self.info, f"'{target}' not in info"
+        self.target = target
+
+    def __getitem__(self, index):
+        return self.raw, self.target
```

Comment:
Is there a use case for returning the entire Raw object as a Raw object? It could be left NotImplemented if there is not.

Reply:
Right now, I removed the transformers from our MOABB and TUH datasets, to get a simple, first running version. In another comment Alex pointed out that we would like to have smart transformers that know whether they are working on raw, continuous data or on epoched data. If they should operate on raw data, this could be a reason to query the raw.

Reply:
I think the idea behind having transformers that can work on Raw or Epochs is so that there is a single class per type of transform (instead of having one for preprocessing on continuous data and one for windowed data).

I realize that this might just be for getting a first running version as you say, but wasn't the WindowsDataset initially meant to apply a (windows) transformer just before returning a window? If so, transformers should be applied inside a Dataset class, and not outside. In that case, I would expect the same thing to happen in a BaseDataset, and so BaseDataset's __getitem__ would first apply a continuous transformer (if necessary) and then return a numpy array instead of a Raw object. What do you think?

Reply:
@gemeinl Reading another one of your answers below made me think: is it because we potentially want to concatenate BaseDatasets directly without passing through mne.Epochs? In that case I would understand the point for a DataFrame (although this might be covered already by the length-1 epoch case we tested during the sprint).